src/precice/impl/ValidationMacros.hpp

```diff
@@ -142,3 +142,20 @@
     const auto id = (dataID);              \
     PRECICE_REQUIRE_DATA_WRITE_IMPL(id)    \
   } while (false)
+
+//
+// DATA VALUE VALIDATION
+//
+#ifdef NDEBUG
+
+#define PRECICE_VALIDATE_DATA(data, size) \
+  {                                       \
+  }
+#else //NDEBUG
+
+#define PRECICE_VALIDATE_DATA(data, size)                                                                                                                                       \
+  do {                                                                                                                                                                          \
```

Comment:
`do { ... } while(false)`  is only required if you want to make a macro behave like a single statement.
As this macro expands to a single statement, this isn't necessary.

```cpp
if (cond)
  CALL_MACRO();

// with do while
if (cond)
  do{ a; b; c; } while(false)

// without do while
if (cond)
  a;
b;
c;
```

A valid use would be if you need to copy the input. Take this as an example:
```cpp
#define CALL_MACRO(input, x) \
do { \
const auto tmp_input = input; \
const auto tmp_x = x; \
do_something(tmp_input, tmp_input+tmp_x); \
while (false)
```

Reply:
I skimmed a bit through the net (before pushing this) in order to understand the necessity.  Some discussion convinced me to still do it with the while false statement, but if you think we should skip, then let's do it.